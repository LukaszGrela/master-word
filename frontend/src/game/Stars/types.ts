export interface IProps {
  score: number;
}
