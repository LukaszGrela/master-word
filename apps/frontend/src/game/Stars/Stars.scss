.stars {
  fill: #979797;

  .gold {
    fill: gold;
  }
  .size-medium {
    width: 4.2rem;
    height: 4.2rem;
    padding-bottom: 0;
  }
  .size-large {
    width: 9rem;
    height: 8.4rem;
  }
}
