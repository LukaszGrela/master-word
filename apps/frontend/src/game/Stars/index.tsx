import { FC } from 'react';

import IconGrade, { TGrade } from '../icons/IconGrade';
import { IProps } from './types';
import { classNames } from '../../utils/classNames';

import './Stars.scss';

const Stars: FC<IProps> = ({ score }) => {
  let grade1: TGrade = 'zero';
  let grade2: TGrade = 'zero';
  let grade3: TGrade = 'zero';

  if (score > 0 && score < 1) {
    grade1 = 'half';
  }
  if (score >= 1) {
    grade1 = 'full';
  }

  if (score > 1 && score < 2) {
    grade2 = 'half';
  }
  if (score >= 2) {
    grade2 = 'full';
  }

  if (score > 2 && score < 3) {
    grade3 = 'half';
  }
  if (score >= 3) {
    grade3 = 'full';
  }

  return (
    <div className="stars">
      <IconGrade
        className={classNames(grade1 !== 'zero' && 'gold', 'size-medium')}
        grade={grade1}
      />
      <IconGrade
        className={classNames(grade2 !== 'zero' && 'gold', 'size-large')}
        grade={grade2}
      />
      <IconGrade
        className={classNames(grade3 !== 'zero' && 'gold', 'size-medium')}
        grade={grade3}
      />
    </div>
  );
};

export default Stars;
