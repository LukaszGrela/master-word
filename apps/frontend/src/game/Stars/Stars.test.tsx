import { describe, expect, it } from 'vitest';
import Stars from './index';
import { render } from '@testing-library/react';

describe('game', () => {
  describe('Stars', () => {
    it.each([0, 0.75, 1, 1.25, 2, 2.5, 3, 42])('Renders score %s', (score) => {
      const { container } = render(<Stars score={score} />);

      expect(container).toMatchSnapshot();
    });
  });
});
