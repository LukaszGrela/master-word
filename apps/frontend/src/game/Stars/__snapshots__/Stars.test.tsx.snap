// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`game > Stars > Renders score 0 1`] = `
<div>
  <div
    class="stars"
  >
    <svg
      class="grade-icon size-medium zero"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
      />
    </svg>
    <svg
      class="grade-icon size-large zero"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
      />
    </svg>
    <svg
      class="grade-icon size-medium zero"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
      />
    </svg>
  </div>
</div>
`;

exports[`game > Stars > Renders score 0.75 1`] = `
<div>
  <div
    class="stars"
  >
    <svg
      class="grade-icon gold size-medium half"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m480-323 126 77-33-144 111-96-146-13-58-136v312ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
    <svg
      class="grade-icon size-large zero"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
      />
    </svg>
    <svg
      class="grade-icon size-medium zero"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
      />
    </svg>
  </div>
</div>
`;

exports[`game > Stars > Renders score 1 1`] = `
<div>
  <div
    class="stars"
  >
    <svg
      class="grade-icon gold size-medium full"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
    <svg
      class="grade-icon size-large zero"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
      />
    </svg>
    <svg
      class="grade-icon size-medium zero"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
      />
    </svg>
  </div>
</div>
`;

exports[`game > Stars > Renders score 1.25 1`] = `
<div>
  <div
    class="stars"
  >
    <svg
      class="grade-icon gold size-medium full"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
    <svg
      class="grade-icon gold size-large half"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m480-323 126 77-33-144 111-96-146-13-58-136v312ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
    <svg
      class="grade-icon size-medium zero"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
      />
    </svg>
  </div>
</div>
`;

exports[`game > Stars > Renders score 2 1`] = `
<div>
  <div
    class="stars"
  >
    <svg
      class="grade-icon gold size-medium full"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
    <svg
      class="grade-icon gold size-large full"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
    <svg
      class="grade-icon size-medium zero"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m354-247 126-76 126 77-33-144 111-96-146-13-58-136-58 135-146 13 111 97-33 143ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Zm247-350Z"
      />
    </svg>
  </div>
</div>
`;

exports[`game > Stars > Renders score 2.5 1`] = `
<div>
  <div
    class="stars"
  >
    <svg
      class="grade-icon gold size-medium full"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
    <svg
      class="grade-icon gold size-large full"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
    <svg
      class="grade-icon gold size-medium half"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m480-323 126 77-33-144 111-96-146-13-58-136v312ZM233-80l65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
  </div>
</div>
`;

exports[`game > Stars > Renders score 3 1`] = `
<div>
  <div
    class="stars"
  >
    <svg
      class="grade-icon gold size-medium full"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
    <svg
      class="grade-icon gold size-large full"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
    <svg
      class="grade-icon gold size-medium full"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
  </div>
</div>
`;

exports[`game > Stars > Renders score 42 1`] = `
<div>
  <div
    class="stars"
  >
    <svg
      class="grade-icon gold size-medium full"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
    <svg
      class="grade-icon gold size-large full"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
    <svg
      class="grade-icon gold size-medium full"
      height="24"
      viewBox="0 -960 960 960"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m233-80 65-281L80-550l288-25 112-265 112 265 288 25-218 189 65 281-247-149L233-80Z"
      />
    </svg>
  </div>
</div>
`;
